pkg/cmd/util/clientcmd/factory.go

```diff
@@ -418,3 +521,30 @@ func (g groupResourcesByName) Less(i, j int) bool {
 	return g[i].Group < g[j].Group
 }
 func (g groupResourcesByName) Swap(i, j int) { g[i], g[j] = g[j], g[i] }
+
+// GetAllPods returns all pods matching the namespace and label selector
+func GetAllPods(client coreclient.PodsGetter, namespace string, selector labels.Selector, sortBy func([]*v1.Pod) sort.Interface) ([]*api.Pod, error) {
```

Comment:
Wouldn't it be better to modify upstream's `GetFirstPod` to watch for pod or not?